ta/trusted_keys/entry.c

```diff
@@ -0,0 +1,326 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (C) 2019-2020, Linaro Limited
+ */
+
+#include <pta_system.h>
+#include <string.h>
+#include <string_ext.h>
+#include <tee_internal_api.h>
+#include <tee_internal_api_extensions.h>
+#include <trusted_keys.h>
+#include <util.h>
+
+#define IV_SIZE			16
+#define TAG_SIZE		16
+#define MAX_BUF_SIZE		512
+
+/*
+ * Acronym:
+ *
+ * TK - Trusted Key
+ */
+
+struct tk_blob_hdr {
```

Comment:
What happens if layout of this struct is changed? Will something break in normal world? That is, is this part of the ABI?

Reply:
Normal world just treats this as a blob with max length as 512 bytes. It doesn't care what's inside it which is something for underlying trust source to understand and parse. BTW, for blob compatibility across multiple OP-TEE versions I have added `reserved` field in case we need to extend the functionality.